python/cudf/cudf/core/column/column.py

```diff
@@ -685,15 +685,32 @@ def _check_scatter_key_length(
                 f"{num_keys}"
             )
 
+    def _validate_fillna_value(
+        self, fill_value: ScalarLike | ColumnLike
+    ) -> cudf.Scalar | ColumnBase:
+        """Align fill_value for .fillna based on column type."""
+        if is_scalar(fill_value):
+            return cudf.Scalar(fill_value, dtype=self.dtype)
+        return as_column(fill_value)
```

Comment:
*question*: use `dtype=self.dtype` here too?

Reply:
I think in theory, yes. I wanted to maintain the old logic here to avoid potential breakage but I can look into this in another PR